plugins/actix-web/src/lib.rs

```diff
@@ -314,3 +351,40 @@ impl actix_web::dev::Factory<(), Ready<Result<HttpResponse, Error>>, Result<Http
         fut_ok(HttpResponse::Ok().json(&*self.0.read()))
     }
 }
+
+#[cfg(feature = "swagger-ui")]
+#[derive(Clone)]
+struct SwaggerUIHandler(String);
+
+#[cfg(feature = "swagger-ui")]
+#[derive(Serialize)]
+struct SwaggerUIContext {
+    api_spec_url: String,
+}
+
+#[cfg(feature = "swagger-ui")]
+static TEMPLATE: &str = include_str!("../templates/swagger-ui.html");
+
+#[cfg(feature = "swagger-ui")]
+impl actix_web::dev::Factory<(), Ready<Result<HttpResponse, Error>>, Result<HttpResponse, Error>>
+    for SwaggerUIHandler
+{
+    fn call(&self, _: ()) -> Ready<Result<HttpResponse, Error>> {
+        let mut tt = TinyTemplate::new();
+        let context = SwaggerUIContext {
+            api_spec_url: self.0.to_owned(),
+        };
+        // This is safe as we are using a given file as input so no errors can occur
+        let _ = tt.add_template("swagger-ui", TEMPLATE);
+
+        let rendered = tt.render("swagger-ui", &context);
```

Comment:
Initializing this here will render it every time we get a request on that path. A more efficient way to do this would be to have a singleton or initialize it when we call `with_swagger_ui_at` and return `Result<Self, Error>` (which happens only once) and we store the rendered string in this struct.